ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtil.java

```diff
@@ -438,7 +463,46 @@ public static void exit_validator(BeaconState state, int index) {
    * @param state - The current BeaconState. NOTE: State will be mutated per spec logic.
    * @param index - The index of the validator that will be penalized.
    */
-  public static void penalize_validator(BeaconState state, int index) {}
+  public static void penalize_validator(BeaconState state, int index) {
+    exit_validator(state, index);
+    Validator validator = state.getValidator_registry().get(index);
+    state
+        .getLatest_penalized_balances()
+        .set(
+            get_current_epoch(state).intValue() % LATEST_PENALIZED_EXIT_LENGTH,
+            state
+                .getLatest_penalized_balances()
+                .get(get_current_epoch(state).intValue() % LATEST_PENALIZED_EXIT_LENGTH)
+                .plus(get_effective_balance(state, index)));
+
+    int whistleblower_index = get_beacon_proposer_index(state, state.getSlot());
+    UnsignedLong whistleblower_reward =
+        get_effective_balance(state, index)
+            .dividedBy(UnsignedLong.valueOf(WHISTLEBLOWER_REWARD_QUOTIENT));
+
```

Comment:
I like how you defined the intermediate variables `whistleblower_index` and `whistleblower_reward`.  Makes it easier to read/debug.